generator/src/main/java/com/scottlogic/deg/generator/walker/ReductiveDecisionTreeWalker.java

```diff
@@ -7,44 +7,46 @@
 import com.scottlogic.deg.generator.generation.ReductiveDataGeneratorMonitor;
 import com.scottlogic.deg.generator.restrictions.RowSpec;
 import com.scottlogic.deg.generator.walker.reductive.FieldCollection;
-import com.scottlogic.deg.generator.walker.reductive.FieldCollectionFactory;
+import com.scottlogic.deg.generator.walker.reductive.FieldCollectionHelper;
 import com.scottlogic.deg.generator.walker.reductive.IterationVisualiser;
 import com.scottlogic.deg.generator.walker.reductive.ReductiveDecisionTreeAdapter;
 
 import java.io.IOException;
 import java.util.stream.Stream;
 
 public class ReductiveDecisionTreeWalker implements DecisionTreeWalker {
-    private final ReductiveDecisionTreeAdapter nodeAdapter = new ReductiveDecisionTreeAdapter();
+    private final ReductiveDecisionTreeAdapter nodeAdapter;
 
     private final IterationVisualiser iterationVisualiser;
-    private final FieldCollectionFactory fieldCollectionFactory;
+    private final FieldCollectionHelper fieldCollectionHelper;
     private final ReductiveDataGeneratorMonitor monitor;
 
     ReductiveDecisionTreeWalker(
         IterationVisualiser iterationVisualiser,
-        FieldCollectionFactory fieldCollectionFactory,
+        FieldCollectionHelper fieldCollectionHelper,
         ReductiveDataGeneratorMonitor monitor) {
         this.iterationVisualiser = iterationVisualiser;
-        this.fieldCollectionFactory = fieldCollectionFactory;
+        this.fieldCollectionHelper = fieldCollectionHelper;
         this.monitor = monitor;
+        this.nodeAdapter = new ReductiveDecisionTreeAdapter(fieldCollectionHelper);
     }
 
     /* initialise the walker with a set (FieldCollection) of unfixed fields */
     public Stream<RowSpec> walk(DecisionTree tree) {
         ConstraintNode rootNode = tree.getRootNode();
-        FieldCollection fieldCollection = fieldCollectionFactory.create(tree);
+        FieldCollection fieldCollection = new FieldCollection(tree.fields);
```

Comment:
???

Reply:
No need for the factory anymore.
FieldCollection is simple enough to not need it.
FieldCollectionHelper is more complex, but its constructor is the same as the factory's constructor, so no need for the middle man class.